File: RecyclingApi.Domain/Entities/Content/Banner.cs
using System;

namespace RecyclingApi.Domain.Entities.Content
{
    /// <summary>
    /// 首页轮播图实体
    /// </summary>
    public class Banner
    {
        /// <summary>
        /// 轮播图ID
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// 轮播图标题
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// 轮播图描述
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// 图片URL
        /// </summary>
        public string ImageUrl { get; set; }

        /// <summary>
        /// 链接地址
        /// </summary>
        public string LinkUrl { get; set; }

        /// <summary>
        /// 排序号（值越小越靠前）
        /// </summary>
        public int Sort { get; set; }

        /// <summary>
        /// 是否启用
        /// </summary>
        public bool IsActive { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// 更新时间
        /// </summary>
        public DateTime? UpdatedAt { get; set; }
    }
} 

File: RecyclingAdmin/src/api/banner.ts
import request from '@/utils/request'

export interface BannerListQueryParams {
  pageindex: number
  pageSize: number
  Keyword: string
  IsActive: boolean
}

export interface BannerData {
  id?: number
  title: string
  description: string
  imageUrl: string
  linkUrl: string
  sort: number
  isActive: boolean
}

/**
 * 获取轮播图列表
 */
export function getBannerList(params: BannerListQueryParams) {
  return request({
    url: '/api/banner/',
    method: 'get',
    params
  })
}

/**
 * 获取轮播图详情
 */
export function getBannerById(id: number) {
  return request({
    url: `/api/banner/${id}`,
    method: 'get'
  })
}

/**
 * 创建轮播图
 */
export function createBanner(data: BannerData) {
  return request({
    url: '/api/banner',
    method: 'post',
    data
  })
}

/**
 * 更新轮播图
 */
export function updateBanner(id: number, data: BannerData) {
  return request({
    url: `/api/banner/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除轮播图
 */
export function deleteBanner(id: number) {
  return request({
    url: `/api/banner/${id}`,
    method: 'delete'
  })
}

/**
 * 切换轮播图状态
 */
export function toggleBannerStatus(id: number) {
  return request({
    url: `/api/banner/${id}/toggle-status`,
    method: 'patch'
  })
} 